Core/android/src/main/java/com/amazonaws/reactnative/core/AWSRNCognitoCredentials.java

```diff
@@ -111,37 +112,51 @@ public void setLogins(final ReadableMap logins) {
 
     @ReactMethod
     public void getIdentityIDAsync(final Promise promise) {
-        try {
-            final String identityId = credentialsProvider.getIdentityId();
-            if (identityId != null) {
-                final WritableMap identityidMap = Arguments.createMap();
-                identityidMap.putString(IDENTITYID, identityId);
-                promise.resolve(identityidMap);
-            }
-        } catch (AmazonServiceException e) {
-            promise.reject(e.getErrorCode(), e.getErrorMessage(), e);
-        } catch (AmazonClientException e) {
-            promise.reject("", e.getMessage(), e);
-        }
+        backgroundRunner.runInBackground(
+            new BackgroundRunner.Supplier<WritableMap>() {
+                @Override
+                public WritableMap get() throws Exception {
+                    final String identityId = credentialsProvider.getIdentityId();
+                    if (identityId != null) {
+                        final WritableMap identityidMap = Arguments.createMap();
+                        identityidMap.putString(IDENTITYID, identityId);
+                        return identityidMap;
+                    }
+                    else {
+                        // TODO: throw exception instead?
+                        return null;
```

Comment:
Note: the existing code doesn't appear to resolve the promise at all in this case. (Same goes for `getCredentialsAsync` below.